testing/sdk_diag.py

```diff
@@ -63,8 +64,12 @@ def get_test_suite_name(item: pytest.Item):
     """Returns the test suite name to use for a given test."""
     # frameworks/template/tests/test_sanity.py => test_sanity_py
     # tests/test_sanity.py => test_sanity_py
-    return os.path.basename(item.parent.name).replace(".", "_")
 
+    # use the class name as the suite name if item is a method
+    if inspect.ismethod(item.obj):
```

Comment:
When tests are part of a class we should use the class name as suite name

Reply:
Sorry @hectorj2f does this mean that there is still work required for this PR. Please re-open if this is the case.

Reply:
Nope, we are good with the merged code :)